### PR TITLE
#108: Setup Alpaca

### PR DESCRIPTION
**Pull request by DPigeon**
# Related Issue
- #108

# Proposed changes
- 3 exposed endpoints in the AlpacaController
- MarketInteractionsFacade builds the proper needed Alpaca objects
- 8 Integration tests on three different endpoints

# Additional Info
- Note: We cannot really test many exceptions or errors in the integration tests since we need a better robust error handling (#122)  and validation strategy before doing so.  
- I also left some TODOs for later. Let me know if something is not clear.

# Checklist (if applicable)
- [x] Integration Tests

# Reviewer(s)
 - @razine-bensari @abdulmansour @wltrf @Alessandro100

### CODE REVIEW

File: backend/src/main/java/com/capstone/moneytree/controller/AlpacaController.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        this.marketInteractionsFacade = marketInteractionsFacade;
+    }
+
+    /**
+     * Get an Alpaca account endpoint.
+     * @return ResponseEntity of Account.
+     */
+    @GetMapping("/account")
+    public ResponseEntity<Account> getAccount() {
+        Account account = marketInteractionsFacade.getAccount();
```
Comment (razine-bensari) [inline backend/src/main/java/com/capstone/moneytree/controller/AlpacaController.java:40]: is the getAccount user level or application level? (we get the account of the user?
)

Comment (DPigeon) [reply backend/src/main/java/com/capstone/moneytree/controller/AlpacaController.java:40]: Yeah we get all of an Alpaca user. https://alpaca.markets/docs/api-documentation/api-v2/account/

Comment (razine-bensari) [reply backend/src/main/java/com/capstone/moneytree/controller/AlpacaController.java:41]: But how do we specify the user?

**Comment by razine-bensari** [comment]
REALLY nice work!, we need to add Java doc in already written and merged code. Thks a lot for adding java doc.

**Review by razine-bensari — APPROVED** [review]

File: backend/src/main/java/com/capstone/moneytree/controller/AlpacaController.java (hunk 1 above)
Comment (DPigeon) [reply backend/src/main/java/com/capstone/moneytree/controller/AlpacaController.java:41]: Alpaca will work with an OAuth ID later when we configure authentication. We'll get the account of a user with its secret key using the auth secret of the properties. Let's discuss about it more next sprint.

**Pull request merged**